contracts/StakingVault/base/RibbonVault.sol

```diff
@@ -0,0 +1,575 @@
+   ***********************************************/
+
+  /// @notice Stores the user's pending deposit for the round
+  mapping(address => Vault.DepositReceipt) public depositReceipts;
+
+  /// @notice On every round's close, the pricePerShare value of sRBN token is stored
+  /// This is used to determine the number of shares to be returned
+  /// to a user with their DepositReceipt.depositAmount
+  mapping(uint256 => uint256) public roundPricePerShare;
+
+  /// @notice Stores pending user withdrawals
+  mapping(address => Vault.Withdrawal) public withdrawals;
+
+  /// @notice Vault's parameters like cap, decimals
+  Vault.VaultParams public vaultParams;
+
+  /// @notice Vault's lifecycle state like round and locked amounts
+  Vault.VaultState public vaultState;
+
+  /// @notice role in charge of weekly vault operations such as buyRBN
+  address public keeper;
+
+  /// @notice Stores addresses of assets of all ribbon vaults
+  address[] public vaultAssets;
+
+  // Gap is left to avoid storage collisions. Though RibbonVault is not upgradeable, we add this as a safety measure.
+  uint256[30] private ____gap;
+
+  // *IMPORTANT* NO NEW STORAGE VARIABLES SHOULD BE ADDED HERE
+  // This is to prevent storage collisions. All storage variables should be appended to RibbonThetaVaultStorage
+  // or RibbonDeltaVaultStorage instead. Read this documentation to learn more:
+  // https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#modifying-your-contracts
+
+  /************************************************
+   *  IMMUTABLES & CONSTANTS
+   ***********************************************/
+
+  /// @notice RBN 0x6123b0049f904d730db3c36a31167d9d4121fa6b
+  address public immutable RBN;
+  /// @notice sRBN
+  ISRBN public immutable sRBN;
+
+  /************************************************
+   *  EVENTS
+   ***********************************************/
+
+  event Deposit(address indexed account, uint256 amount, uint256 round);
+
+  event InitiateWithdraw(
+    address indexed account,
+    uint256 shares,
+    uint256 round
+  );
+
+  event Redeem(address indexed account, uint256 share, uint256 round);
+
+  event CapSet(uint256 oldCap, uint256 newCap, address manager);
+
+  event Withdraw(address indexed account, uint256 amount, uint256 shares);
+
+  /************************************************
+   *  CONSTRUCTOR & INITIALIZATION
+   ***********************************************/
+
+  /**
+   * @notice Initializes the contract with immutable variables
+   * @param _rbn is the RBN contract
+   * @param _srbn is the sRBN contract
+   */
+  constructor(address _rbn, address _srbn) {
+    require(_rbn != address(0), "!_rbn");
+    require(_srbn != address(0), "!_srbn");
+
+    RBN = _rbn;
+    sRBN = ISRBN(_srbn);
+  }
+
+  /**
+   * @notice Initializes the Vault contract with storage variables.
+   */
+  function baseInitialize(
+    address _owner,
+    address _keeper,
+    address[] memory _vaultAssets,
```

Comment:
Since we're just reading the values from `vaultAssets` by index, we can change this to `calldata`